static/src/javascripts/projects/common/modules/onward/breaking-news.js

```diff
@@ -32,137 +32,197 @@ define([
+    function pickNewest(alerts) {
+        return alerts.sort(function (a, b) {
+            return b.frontPublicationDate - a.frontPublicationDate;
+        })[0];
+    }
+
+    // show an alert
+    function alert(alert) {
+        if (alert) {
+            var $body = bonzo(document.body);
+            var $breakingNews = bonzo(qwery('.js-breaking-news-placeholder'));
+            var trackingMessage = 'breaking news alert shown' + (has(knownAlertIDs, alert.id) ? '' : ' 2 or more times');
+
+            // if its the first time we've seen this alert, we wait 3 secs to show it
+            // otherwise we show it immediately
+            var alertDelay = has(knownAlertIDs, alert.id) ? 0 : 3000;
+
+            // $breakingNews is hidden, so this won't trigger layout etc
+            $breakingNews.append(renderAlert(alert));
+
+            // copy of breaking news banner (with blank content) used inline at the
+            // bottom of the body, so the bottom of the body can visibly scroll
+            // past the pinned alert
+            var $spectre = renderSpectre($breakingNews);
+
+            // inject the alerts into DOM
+            setTimeout(function () {
+                fastdom.write(function () {
+                    if (alertDelay === 0) {
+                        $breakingNews.removeClass('breaking-news--fade-in');
+                    }
+                    $body.append($spectre);
+                    $breakingNews.removeClass('breaking-news--hidden');
+                    markAlertAsSeen(alert.id);
+                });
+                omniture.trackLink(this, trackingMessage);
+            }, alertDelay);
+        }
+        return alert;
+    }
+
+    function renderAlert(alert) {
+        alert.marque36icon = svgs('marque36icon');
+        alert.closeIcon = svgs('closeCentralIcon');
+
+        var $alert = bonzo.create(template(alertHtml, alert));
+
+        bean.on($('.js-breaking-news__item__close', $alert)[0], 'click', function () {
+            fastdom.write(function () {
+                $('[data-breaking-article-id]').hide();
+            });
+            markAlertAsDismissed(alert.id);
+        });
+
+        return $alert;
+    }
+
+    function renderSpectre($breakingNews) {
+        return bonzo(bonzo.create($breakingNews[0]))
+            .addClass('breaking-news--spectre')
+            .removeClass('breaking-news--fade-in breaking-news--hidden');
+    }
+
+    return function () {
+        if (userCanDismissAlerts()) {
+            knownAlertIDs = storage.local.get(knownAlertIDsStorageKey) || {};
+
+            return fetchBreakingNews()
+                .then(parseResponse)
+                .then(getRelevantAlerts)
+                .then(pruneKnownAlertIDs)
+                .then(filterAlertsByDismissed)
+                .then(filterAlertsByAge)
+                .then(pickNewest)
+                .then(alert);
+        } else {
+            return Promise.reject('cannot dismiss');
```

Comment:
A rejected promise is the same thing as an async error. If this isn't an error maybe this function should just return nothing instead of a rejected promise?